torch/csrc/jit/pybind_utils.h

```diff
@@ -21,6 +21,9 @@
 #include <torch/csrc/utils/auto_gil.h>
 #include <torch/csrc/utils/pybind.h>
 #include <torch/csrc/utils/six.h>
+#ifdef USE_DISTRIBUTED
```

Comment:
What's the reason for this macro? It seems weird to have the checks here but nowhere else…if we weren't compiling with the distributed package, how would the rest of it work?

Reply:
Yeah the pickling support for rref is covered by @zhaojuanmao PR https://github.com/pytorch/pytorch/pull/32959. 

For the reason of this macro, I think right now default libtorch is not shipped together with distributed support, so everything for the distributed package get built guarded by this `USE_DISTRIBUTED` macro. cc @pritamdamania @mrshenli for the exact reason why we not default building the distributed package yet. 

If we weren't compiling with the distributed package, we will not include the corresponding header and JIT will just error out to user saying "RRef is only supported with the distributed package"